### PR TITLE
docs: add data onboarding guide

### PR DESCRIPTION
**Pull request by patheard**
# Summary
Add a guide explaining how to add new datasets to the lake 🌊 

[Rendered view](https://github.com/cds-snc/data-lake/blob/docs/onboarding/docs/data/onboarding.md)

# Related
- Closes https://github.com/cds-snc/platform-core-services/issues/652

### CODE REVIEW

File: docs/data/onboarding.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Onboarding new datasets
+
+Adding new datasets to the data lake usually involves three steps:
+
+1. Add the new data to the [`Raw` bucket](https://github.com/cds-snc/data-lake/blob/main/terragrunt/aws/buckets/raw.tf).
```
Comment (wmoussa-gc) [inline docs/data/onboarding.md:5]: should we mention it should be `unclassified data` (until we get an ATO for PB) ?

**Review by willeybryan — APPROVED** [review]
lgtm

**Review by wmoussa-gc — CHANGES_REQUESTED** [review]

File: docs/data/onboarding.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+### Define a Glue ETL job
+
+Extract, Transform, Load (ETL) jobs are used to make the raw data ready for analysis.  This includes operations like:
+
+- joining data sources to enrich the data,
+- updated column data types,
+- redacting sensitive information, and
+- removing data that isn't useful for analysis.
+
+We are using [Glue Visual ETL](https://docs.aws.amazon.com/glue/latest/dg/author-job-glue.html) to create and manage our jobs in the AWS console.  [A nightly GitHub workflow](https://github.com/cds-snc/data-lake/blob/main/.github/workflows/glue_job_sync.yml) then handles synchronizing the job source back to this [GitHub repository's `/terragrunt/aws/glue/etl` folder](https://github.com/cds-snc/data-lake/tree/main/terragrunt/aws/glue/etl).
```
Comment (wmoussa-gc) [inline docs/data/onboarding.md:76]: should we talk about giving access to the console? should we mentioned that the data lake is one instance only (for our prod and staging BI tool)

File: docs/data/onboarding.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+The ETL job should perform the following two tasks:
+
+1. Write the transformed dataset to the S3 `Transformed` bucket.  The folder structure should match the dataset's `Raw` folder structure.
+2. Update the dataset's Glue Catalog `Transformed` schema.
+
```
Comment (wmoussa-gc) [inline docs/data/onboarding.md:82]: perhaps add that the jobs are automatically monitored?

File: docs/data/onboarding.md (hunk 1 above)
Comment (patheard) [reply docs/data/onboarding.md]: Good idea - I've updated the doc to mention that.

File: docs/data/onboarding.md (hunk 2 above)
Comment (patheard) [reply docs/data/onboarding.md]: Makes sense - I've added a link to our repo to request access as needed and a note in the introduction of the doc talking about the data lake and BI tool environments.

**Review by wmoussa-gc — APPROVED** [review]
lgtm

**Pull request merged**